core/stream/convert.go

```diff
@@ -290,29 +299,59 @@ func intCollapse(count int, dst, src buf) error {
 	sourceStream := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
 	destStream := binary.BitStream{Data: dst.bytes, WritePos: dst.offset}
 	maxPossibleValue := math.Pow(2, float64(srcBitsExcSign)) - 1
+	maxWithSign := math.Pow(2, float64(srcBitsIncSign)) - 1
 	min, max := maxPossibleValue, float64(0)
 	if src.component.Channel == Channel_Depth {
 		for i := 0; i < count; i++ {
-			f := float64(sourceStream.Read(srcBitsExcSign)) / maxPossibleValue
+			f := float64(0)
+
+			if srcTy.Signed && !signed {
+				f = float64(sourceStream.Read(srcBitsIncSign))
+				f = (f + ((maxWithSign + 1) / 2)) / maxWithSign
+			} else {
+				f = float64(sourceStream.Read(srcBitsExcSign)) / maxPossibleValue
+			}
+
 			if f < min {
 				min = f
 			}
 			if f > max {
 				max = f
 			}
+			sourceStream.ReadPos += src.stride - srcBitsIncSign
 		}
 		sourceStream.ReadPos = src.offset
 	}
 	for i := 0; i < count; i++ {
-		v := sourceStream.Read(srcBitsExcSign)
+		v := uint64(0)
```

Comment:
Readability: I wonder if it would be possible to have a separate function for each case rather than lot's of if together, or instead of checking signed condition everywhere, just checking it in the beginning and doing the rest under that single if. e.g. moving the channel check under the first if for both cases.

Reply:
I'll need a second opinion on this.

Reply:
Now that I think about it, I might be able to get rid of the of checks inside the depth channel block since I don't think there are any signed integer depth formats.